flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/TaskExecutorITCase.java

```diff
@@ -38,42 +38,50 @@
 import org.apache.flink.runtime.testutils.CommonTestUtils;
 import org.apache.flink.runtime.testutils.MiniClusterResource;
 import org.apache.flink.runtime.testutils.MiniClusterResourceConfiguration;
-import org.apache.flink.util.TestLogger;
 import org.apache.flink.util.function.SupplierWithException;
 
-import org.junit.Rule;
-import org.junit.Test;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
 
 import java.io.IOException;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CountDownLatch;
 import java.util.function.Predicate;
 import java.util.function.Supplier;
 
-import static org.hamcrest.Matchers.is;
-import static org.junit.Assert.assertThat;
+import static org.assertj.core.api.Assertions.assertThat;
 
 /** Integration tests for the {@link TaskExecutor}. */
-public class TaskExecutorITCase extends TestLogger {
+class TaskExecutorITCase {
 
     private static final int NUM_TMS = 2;
     private static final int SLOTS_PER_TM = 2;
     private static final int PARALLELISM = NUM_TMS * SLOTS_PER_TM;
 
-    @Rule
-    public final MiniClusterResource miniClusterResource =
+    private final MiniClusterResource miniClusterResource =
```

Comment:
Hi @1996fanrui.
I noticed that we can only depend on `InternalMiniClusterExtension` in flink-runtime module, because there's no dependency of `flink-test-utils`.
The `InternalMiniClusterExtension` also holds MiniClusterResource but has no Before-After-Each hooks.
Any suggestions of this?

Reply:
Hi @1996fanrui.
I refactor the code by using `InternalMiniClusterExtension` and `EachCallBackWrapper` and it works fine.
Please help review it again when you have time.
Thanks a lot.

```java
    private static final InternalMiniClusterExtension MINI_CLUSTER_EXTENSION =
            new InternalMiniClusterExtension(
                    new MiniClusterResourceConfiguration.Builder()
                            .setNumberTaskManagers(NUM_TMS)
                            .setNumberSlotsPerTaskManager(SLOTS_PER_TM)
                            .build());

    @RegisterExtension
    private final EachCallbackWrapper<?> miniClusterExtensionWrapper =
            new EachCallbackWrapper<>(
                    new CustomExtension() {
                        @Override
                        public void before(ExtensionContext context) throws Exception {
                            MINI_CLUSTER_EXTENSION.beforeAll(context);
                        }

                        @Override
                        public void after(ExtensionContext context) throws Exception {
                            MINI_CLUSTER_EXTENSION.afterAll(context);
                        }
                    });
```

Reply:
Sounds make sense to me.